### PR TITLE
Backport of 4 patches from RH7

### PR DESCRIPTION
**Pull request by johnsongeorge-w**
@santoshx @prashantmhetar 
This changes required to be merged separately. Don't squash merge.

While reviewing backport - "RH6: Drivers: hv_netvsc: fix network namespace issues with VF support" you could see that I have not patched "get_netvsc_byref()". This is as per the discussion with Dexuan. I have forwarded an email, please check.

### CODE REVIEW

**Comment by santoshx** [comment]
Have you backport these changes from upstream or RH7?
Looks like from RH7, hence commit message needs correction.
Also separate each commit into separate PR for easy review.

File: hv-rhel6.x/hv/netvsc_drv.c [hunk 1]
```diff
@@ -126,65 +134,67 @@ static int netvsc_open(struct net_device *net)
...
+		u32 aread = 0;
 		for (i = 0; i < nvdev->num_chn; i++) {
-			chn = nvdev->chan_table[i].channel;
+			struct vmbus_channel *chn
+				= nvdev->chan_table[i].channel;
+
 			if (!chn)
 				continue;
 
-			aread = hv_get_bytes_to_read(&chn->inbound);
```
Comment (santoshx) [inline hv-rhel6.x/hv/netvsc_drv.c]: I don't see removal of these lines in RH7 patch

**Comment by santoshx** [comment]
There seems to be some problem with "remove useless api" backport.
I can't find the RH7 commit 72d0a19a05b75431b6993a045d44fac6af7d9b26 mentioned in RH7.

**Comment by santoshx** [comment]
Please insert lines in commit messages appropriately.

**Comment by johnsongeorge-w** [comment]
@santoshx I will use the RH7 commit to validate and create a new PR to make the changes clear. I will push each change in each branch to make the changes clearer. Closing this PR to start a fresh one.

**Pull request closed without merging**